### ISSUE TITLE
PT007 shows wrong type in violation message

### ISSUE THREAD

**Issue by ThiefMaster** [astral-sh/ruff#14743]
The violation message shows the same type twice, even though one of them should be `list`.

```python
import pytest


@pytest.mark.parametrize('foo', [{'bar': 1}, {'baz': 1}])
def test_test(foo):
    pass
```

```sh
$ ruff version
ruff 0.8.1
```

```py
$ ruff check --isolated --select PT007 --config "lint.flake8-pytest-style.parametrize-values-type = 'tuple'" --preview --no-cache rufftest/ruff_sample.py
rufftest/ruff_sample.py:4:33: PT007 Wrong values type in `@pytest.mark.parametrize` expected `tuple` of `tuple`
  |
4 | @pytest.mark.parametrize('foo', [{'bar': 1}, {'baz': 1}])
  |                                 ^^^^^^^^^^^^^^^^^^^^^^^^ PT007
5 | def test_test(foo):
6 |     pass
  |
  = help: Use `tuple` of `tuple` for parameter values

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

The fix looks fine though so this is just a wrong message:

```py
$ ruff check --isolated --select PT007 --config "lint.flake8-pytest-style.parametrize-values-type = 'tuple'" --preview --no-cache rufftest/ruff_sample.py --diff --unsafe-fixes
--- rufftest/ruff_sample.py
+++ rufftest/ruff_sample.py
@@ -1,6 +1,6 @@
 import pytest


-@pytest.mark.parametrize('foo', [{'bar': 1}, {'baz': 1}])
+@pytest.mark.parametrize('foo', ({'bar': 1}, {'baz': 1}))
 def test_test(foo):
     pass

Would fix 1 error.
```

**Comment by charliermarsh** [comment]
I think this is right, but perhaps unclear -- it's saying that it "expected a tuple that contains tuples", i.e., `((`.

**Comment by ThiefMaster** [comment]
Oh indeed, I read this "tuple of tuple" as "tuple instead of tuple"...

**Comment by charliermarsh** [comment]
Yeah I think we can make it clearer...

**Comment by ThiefMaster** [comment]
FWIW the real code where this happened was this:

```py
def setup_fixtures(func):
    """Set up fixtures (utlity decorator)."""
    func = pytest.mark.usefixtures('smtp', 'mock_access_request', 'dummy_access_request')(func)
    return pytest.mark.parametrize('mock_access_request', [{
        'during_registration': True,
        'during_registration_required': True,
        'personal_data': generate_personal_data(),
        'include_accompanying_persons': False
    }, {
        'during_registration': True,
        'during_registration_required': True,
        'personal_data': generate_personal_data(),
        'include_accompanying_persons': True
    }], indirect=True)(func)
```

I guess with `indirect=True` parametrization, but also with any case where there's only a single param, the error message is particularly strange. I haven't tested it, but I guess w/a single param the actual param value inside the outer tuple can be anything (not sure what happens if you use a tuple/list there though).

**Comment by harupy** [comment]
Can we remove `of {expected_row_type}` for a single param?

```python
$ ruff check --isolated --select PT007 --config "lint.flake8-pytest-style.parametrize-values-type = 'tuple'" --preview --no-cache rufftest/ruff_sample.py
rufftest/ruff_sample.py:4:33: PT007 Wrong values type in `@pytest.mark.parametrize` expected `tuple`

  |
4 | @pytest.mark.parametrize('foo', [{'bar': 1}, {'baz': 1}])
  |                                 ^^^^^^^^^^^^^^^^^^^^^^^^ PT007
5 | def test_test(foo):
6 |     pass
  |
  = help: Use `tuple` of `tuple` for parameter values

Found 1 error.
No fixes available (1 hidden fix can be enabled with the `--unsafe-fixes` option).
```

**Comment by BitWeaverDev** [comment]
Hi! I was looking at this issue and I think I understand what needs to be changed. Is anyone currnetly working on it? I would like to take a shot at fixing this.

From what I can see, the message `Wrong values type in `@pytest.mark.parametrize` expected `tuple` of `tuple`` is confusing when there's only one parameter — because the outer container doesn't have rows in that case, the `of {row}` part doesn't really make sense. Also when both the values and row types are the same (like both are `tuple`), the message looks like a duplicate even when it's actualy correct.

I already found the relevant code in `parametrize.rs` and I think the fix would be to either adjust the message when there's single parameter, or at least make the wording clearer so users can understand what the rule is actually telling them.

Let me know if this is still available to work on!
